### PR TITLE
Update to VST 2.12

### PR DESCRIPTION
**Pull request by roconnor-blockstream**

### CODE REVIEW

**Comment by roconnor-blockstream** [comment]
@uncomputable can you test out the manual build instructions?  If you look at the `.nix` files you can see that I had to engage `IGNORECOMPCERTVERSION=true` to get this all to work.  However, I suspect that if one uses `opam` they will not have this problem.

**Review by uncomputable — COMMENTED** [review]
b3351f04846a1c75299b9a387613aa6235d883a9: The changes look good and build on my machine. I will try the manual build next.

File: README.md [hunk 1]
```diff
@@ -64,8 +64,8 @@ Packages in the Coq ecosystem are managed by the [opam package manager](https://
 
 We need a custom build and **cannot** use opam for this step.
 
-1. `wget -O - https://github.com/PrincetonUniversity/VST/archive/v2.11.tar.gz | tar -xvzf -`
-1. `cd VST-2.11`
+1. `wget -O - https://github.com/PrincetonUniversity/VST/archive/v2.12.tar.gz | tar -xvzf -`
+1. `cd VST-2.12`
 1. `make -j$(nproc) default_target sha`
```
Comment (uncomputable) [inline README.md:69]:
```
$ make -j$(nproc) default_target sha
Makefile:131: *** COMPCERT VERSION MISMATCH: COMPCERT_VERSION=version=3.12 but /home/user/.opam/default/lib/coq/user-contrib/compcert/VERSION=version=3.11.  Stop.
```

I had ignore this conflict, like in nix. The compcert version seems to be fixed to the Coq version, so I don't see another workaround.

```
export IGNORECOMPCERTVERSION=true
```

**Review by uncomputable — CHANGES_REQUESTED** [review]
b3351f04846a1c75299b9a387613aa6235d883a9: Let's update the README. The rest looks good. I successfully built the Coq project manually.

**Comment by uncomputable** [comment]
Everything builds on my end.

**Review by uncomputable — APPROVED** [review]
ACK dedac3ed39c13eccf6f956bd648d20e6a316a1f5

**Pull request merged**